pkg/kubeapiserver/options/authentication.go

```diff
@@ -654,18 +664,27 @@ func (o *BuiltInAuthenticationOptions) ApplyTo(ctx context.Context, authInfo *ge
 	authInfo.Authenticator = authenticator
 
 	if len(o.AuthenticationConfigFile) > 0 {
+		authenticationconfigmetrics.RegisterMetrics()
 		trackedAuthenticationConfigData := authenticatorConfig.AuthenticationConfigData
 		var mu sync.Mutex
 		go filesystem.WatchUntil(
 			ctx,
 			time.Minute,
 			o.AuthenticationConfigFile,
 			func() {
-				// TODO add metrics
 				// TODO collapse onto shared logic with DynamicEncryptionConfigContent controller
+				var (
+					err        error
+					authConfig *apiserver.AuthenticationConfiguration
+				)
 
 				mu.Lock()
-				defer mu.Unlock()
```

Comment:
keep the unlock defer on its own (makes it easier to read and reason about interaction if something else in a defer panics)